### PR TITLE
Skip and fix tests for io.js

### PR DESCRIPTION
**Pull request by arv**
This skips and updates the tests for IO.js 1.2

### CODE REVIEW

File: test/feature/Modules/Error_InvalidModuleDeclaration.module.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Error: File not found 'test/feature/Modules/resources/no_such_file.js'
+// Error: 'test/feature/Modules/resources/no_such_file.js'
```
Comment (arv) [inline test/feature/Modules/Error_InvalidModuleDeclaration.module.js]: The IO.js message was:

```
Error: ENOENT: no such file or directory, open 'test/feature/Modules/resources/no_such_file.js'
```

The current substring is kind of sad though.

**Comment by arv** [comment]
@johnjbarton PTAL

**Comment by johnjbarton** [comment]
I don't understand the motivation here. It seems like we are removing tests that io.js cannot pass. Is that correct?

**Comment by arv** [comment]
These are bugs in V8.

The motivation is to be able to use Node 0.12 or io.js on Travis.

**Comment by domenic** [comment]
An alternative is the es6-shim approach which will replace the native versions with their shims if the native versions exhibit bugs.

**Comment by arv** [comment]
Yeah, but these are such edge cases. To trigger the -0 bug you need to have a Map/Set with 8+ elements and do a lookup based on -0. Since the traceur test was the first case where this was found I'm not sure it is worth adding startup runtime cost to detect.

**Comment by arv** [comment]
@johnjbarton Are you convinced yet?

**Comment by johnjbarton** [comment]
yup, LGTM :-/
